### ISSUE TITLE
Use Dataflow Flex Templates and remove use of ValueProvider

### ISSUE THREAD

**Issue by jklukas** [mozilla/gcp-ingestion#1534]
[Dataflow Flex templates](https://cloud.google.com/dataflow/docs/concepts/dataflow-templates) are now GA. They allow publishing a docker container that compiles the pipeline at runtime so that any configuration parameter can be specified at runtime, not just those that are marked as ValueProvider.

This would allow us to clean up the code a bit, and would provide more flexibility in how we define new parameters in the future. There are a few [documented downsides](https://cloud.google.com/dataflow/docs/concepts/dataflow-templates#comparing-templated-jobs) which at time of writing are lack of support for FlexRS (which we have not so far used) and lack of support for updating streaming jobs (which we also don't do).

This may not be worth prioritizing until we have some clear motivation to do so. It would require a close collaboration between data eng and ops, since it completely changes the workflow for publishing a template and deploying a job.

cc @whd

**Comment by relud** [comment]
This is going to be needed for Pub/Sub Lite IO, because it doesn't support ValueProvider.

Also, flex templates are a different resource type in terraform: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/dataflow_flex_template_job

**Comment by whd** [comment]
It looks like we already have a use case and flex templates was on :jason's stretch goals to investigate last year.

This does completely change the workflow and will require coordination, but that coordination will be necessary for the use case being investigated anyway (pubsub lite). Of particular note, we're using a custom build of terraform 0.11 for beam and shared-prod right now. Moving to modern terraform is planned for this quarter, but is currently blocked on https://bugzilla.mozilla.org/show_bug.cgi?id=1682749 for data-shared. Almost assuredly upstream terraform by now supports the custom options we added to our custom provider for beam, but I will need to go through and verify that as part of the update process here.

As a side note: :jason and I confirmed last year that the standard dataflow resource in modern terraform *does* support updates operations now, and we've used it to effect in the opsec pipeline. For the purposes of the standard ingestion pipeline on 0.11 however, moving to flex templates should be equivalent to the current system.

**Comment by jklukas** [comment]
Also note that Beam supports Java 11 as of a recent release (I believe 2.29). It may make sense to roll this into our change to flex template deploys, using the Java 11 base image.

**Comment by whd** [comment]
This is being tracked in [DSRE-79](https://mozilla-hub.atlassian.net/browse/DSRE-79).

**Comment by whd** [comment]
https://github.com/mozilla-services/cloudops-infra/pull/3293 landed today so I believe the code cleanup work described in this issue can commence. We probably want to remove the old `build-template` script etc. as well since it won't be used going forward.

Another thing to note is that per https://cloud.google.com/dataflow/docs/concepts/dataflow-templates#comparing-templated-jobs flex templates support all things now including job updates; I've filed https://mozilla-hub.atlassian.net/browse/DSRE-110 to track testing that out.
